src/rpc/blockchain.cpp

```diff
@@ -428,7 +428,8 @@ static RPCHelpMan getblockfrompeer()
         "getblockfrompeer",
         "Attempt to fetch block from a given peer.\n\n"
         "We must have the header for this block, e.g. using submitheader.\n"
-        "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n\n"
+        "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n"
+        "Note: On a pruned node this block can be pruned again as soon as the prune height surpasses the block height at the time it was fetched.\n\n"
```

Comment:
If a user calls getblockfrompeer 200 times or so, it's quite possible they will fill a block file and get pruned *before* the current height is pruned.

The only way to guarantee any survival of a fetched block would be something like #19463

Reply:
Hm, that's correct, how would you like `Note: There is no guarantee the block will be available at a specific time in the future as it may be gone with the next pruning event` ?

Reply:
Maybe a bit long. How about "Note: The block could be re-pruned as soon as it is received."

Though if the peer simply never responds, it's basically the same outcome?

Reply:
Thanks, I took your suggested test.

And it's true, it's the same outcome if the block is never received. But I don't think we have to worry about that here, the wording above says "Attempt to fetch..." which implies the operation might fail and there should be logging in place as well.